### DISCUSSION TITLE
Password with confirmation cannot work

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6421] [Help] [answered]
This may be very easy for others, but I cannot make working..

Form:

```
Forms\Components\TextInput::make('password')
    ->dehydrateStateUsing(fn ($state) => bcrypt($state))
    ->required()
    ->minLength(8)
    ->maxLength(255)
    ->confirmed()
    ->password(),

Forms\Components\TextInput::make('passwordConfirmation')
    ->maxLength(255)
    ->required()
    ->password(),
```

Always getting validation error `The password field confirmation does not match.` if I enter same or different password.. any idea?

**Edited**

`->same('passwordConfirmation')` also not working

**Comment by ArtMin96** [comment]
Do you have the `passwordConfirmation` property in the class?

```php
public null | string $passwordConfirmation = '';
```

**Reply by valpuia** [reply]
Yes, I put like this `public $passwordConfirmation;` and also for password `public $password;`

**Comment by valpuia** [comment, marked as answer]
I have made it working, all I have to do is put the confirmation name as `password_confirmation` not any other variable and it's working.
They follow the Laravel `confirmed` [https://laravel.com/docs/10.x/validation#rule-confirmed](https://laravel.com/docs/10.x/validation#rule-confirmed)
